crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -90,8 +93,27 @@ impl AccountBuilder {
         self
     }
 
+    /// Adds a designated authentication [`AccountComponent`] to the builder.
+    /// This component must have exactly one procedure.
+    ///
+    /// This component will be placed at index 0 of the account procedures list.
+    pub fn with_auth_component(mut self, account_component: impl Into<AccountComponent>) -> Self {
+        let component = account_component.into();
+
+        // Validate that the component has exactly one procedure
+        let procedure_count = component
+            .library()
+            .module_infos()
+            .map(|module| module.procedures().count())
+            .sum::<usize>();
+        assert_eq!(procedure_count, 1, "auth component must have exactly one procedure");
```

Comment:
Instead of checking the number of procedures directly, it may be a good idea to add something like `AccountComponent::is_auth_component()` method and use that to check the correctness of the operation. This way, it would be easier to update how we define a valid auth component down the road.

Reply:
also in [213857c](https://github.com/0xMiden/miden-base/pull/1480/commits/213857c663ae8f71aed0b2711361c33f30ee51f0)